### PR TITLE
chore: convert Jest tests to Karma/Mocha/Chai in fast-web-utilities

### LINKED ISSUE #4548: Update `fast-web-utilities` test dependencies to rely on mocha and chai
# Description
Similar to #4546.

Remove `jest` as a dependency for `fast-web-utilities` and rely on `mocha` and `chai` to align testing strategies with the `web-components` package.

## Requirements

- Remove `jest` as a dependency for `fast-web-utilities`
- Add `mocha` and `chai` as devDependencies
- Update syntax for `.spec` files to reflect this change

### PR DESCRIPTION
**Pull request by janechu**
# Pull Request

## 📖 Description

This test converts all `Jest` tests found in the `@microsoft/fast-web-utilities` package to the Karma/Mocha/Chai test setup used in the web-components packages.
 
### 🎫 Issues

Closes #4548 

## 👩‍💻 Reviewer Notes

There were a very few issues with the conversion, namely the use of Jests stored arguments. In one case this was a little different I checked the use of `nonce` with a `MutationObserver` which can be found in `dom.spec.ts` on line 220.

### General

- [x] I have included a change request file using `$ yarn change`
- [x] I have added tests for my changes.
- [x] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/community/code-of-conduct/#our-standards) for this project.

### Component-specific

- [ ] I have added a new component
- [ ] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

### CODE REVIEW

**Review by nicholasrice — APPROVED** [review]

**Review by awentzel — APPROVED** [review]

File: packages/utilities/fast-web-utilities/src/dom.spec.ts [hunk 1]
```diff
@@ -9,206 +11,215 @@ import {
...
-                new KeyboardEvent("keypress", {
-                    which: 40,
-                } as KeyboardEventInit)
-            )
-        ).toBe(KeyCodes.arrowDown);
+            getKeyCode({
+                which: 40,
+                keyCode: 40,
+                charCode: 40,
+            } as any)
```
Comment (EisenbergEffect) [inline packages/utilities/fast-web-utilities/src/dom.spec.ts:132]: Out of curiosity..why the change to `any` and the move away from `KeyboardEvent`?

**Review by EisenbergEffect — APPROVED** [review]

**Review by chrisdholt — APPROVED** [review]

File: packages/utilities/fast-web-utilities/src/dom.spec.ts (hunk 1 above)
Comment (janechu) [reply packages/utilities/fast-web-utilities/src/dom.spec.ts:132]: Because `KeyboardEvent` was giving me type errors since `which` is deprecated.
